engine/src/main/java/com/agiletec/aps/system/common/AbstractSearcherDAO.java

```diff
@@ -32,464 +32,469 @@
+    }
+
+    protected void addObjectSearchStatementBlock(PreparedStatement stat,
+                                                 int index,
+                                                 Object object,
+                                                 Integer dateDelay,
+                                                 boolean isLikeOption) throws SQLException {
+        this.addObjectSearchStatementBlock(stat, index, object, dateDelay, isLikeOption, null);
+    }
+
+    protected void addObjectSearchStatementBlock(PreparedStatement stat,
+                                                 int index,
+                                                 Object object,
+                                                 Integer dateDelay,
+                                                 boolean isLikeOption,
+                                                 FieldSearchFilter.LikeOptionType likeOptionType) throws SQLException {
+
+        if (object instanceof String) {
+            if (isLikeOption) {
+                object = ((String) object).toUpperCase();
+                String parameter = null;
+                if (null == likeOptionType || likeOptionType.equals(FieldSearchFilter.LikeOptionType.COMPLETE)) {
+                    parameter = "%" + ((String) object) + "%";
+                } else if (likeOptionType.equals(FieldSearchFilter.LikeOptionType.LEFT)) {
+                    parameter = "%" + ((String) object);
+                } else if (likeOptionType.equals(FieldSearchFilter.LikeOptionType.RIGHT)) {
+                    parameter = ((String) object) + "%";
+                }
+                stat.setString(index, parameter);
+            } else {
+                stat.setString(index, (String) object);
+            }
+        } else if (object instanceof Date) {
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTime((Date) object);
+            if (dateDelay != null) {
+                calendar.add(Calendar.DATE, dateDelay);
+            }
+            if (object instanceof Timestamp) {
+                Timestamp timestamp = new Timestamp(calendar.getTime().getTime());
+                stat.setTimestamp(index, timestamp);
+            } else {
+                Date data = calendar.getTime();
+                stat.setDate(index, new java.sql.Date(data.getTime()));
+            }
+        } else if (object instanceof BigDecimal) {
+            stat.setBigDecimal(index, (BigDecimal) object);
+        } else if (object instanceof Boolean) {
+            stat.setString(index, ((Boolean) object).toString());
+        } else {
+            stat.setObject(index, object);
+        }
+    }
+
+    protected String createQueryString(FieldSearchFilter[] filters, boolean isCount, boolean selectAll) {
+        StringBuffer query = this.createBaseQueryBlock(filters, isCount, selectAll);
+
+        boolean hasAppendWhereClause = false;
+        if (isCount) {

```

Comment:
This if statement isn't needed. The call to 
appendMetadataFieldFilterQueryBlocksForCount
and
appendMetadataFieldFilterQueryBlocksFor

End up making the same filter calls